src/conversions/std/num.rs

```diff
@@ -738,4 +748,14 @@ mod tests {
     test_nonzero_common!(nonzero_usize, NonZeroUsize);
     test_nonzero_common!(nonzero_i128, NonZeroI128);
     test_nonzero_common!(nonzero_u128, NonZeroU128);
+
+    #[test]
+    fn test_i64_bool() {
+        Python::with_gil(|py| {
+            let obj = true.to_object(py);
+            assert_eq!(1, obj.extract::<i64>(py).unwrap());
+            let obj = false.to_object(py);
+            assert_eq!(0, obj.extract::<i64>(py).unwrap());
+        })
+    }
```

Comment:
It might be worth adding a test that a float does not covert to i32 or i64, which I believe was the motivation of calling `PyNumber_Index` directly.

Reply:
added `i64` tests, for float, is that enough?